src/main/java/org/example/quizMates/service/impl/PairServiceImpl.java

```diff
@@ -13,28 +14,33 @@
 public class PairServiceImpl implements PairService {
     private final PairRepository pairRepository;
 
+    private final static String PAIR_NOT_FOUND= "Pair with id %s not found";
+
     @Override
-    public Pair findById(Long aLong) {
-        return null;
+    public Pair findById(Long id) {
+        return pairRepository.findById(id).orElseThrow(
+                () -> new ResourceNotFoundException(String.format(PAIR_NOT_FOUND)));
```

Comment:
Pass your id as second argument to the String.format() method so that client can get message like - Pair with id 1 not found

Reply:
Done